### PR TITLE
BUGFIX: Enforce unique names also for root children

### PR DESCRIPTION
**Pull request by nezaniel**
Until now, we checked uniqueness for node names for an explicitly given origin of the parent.
While this makes sense in general, the constraint checks fails for more than zero dimensions when the parent is a root node, because root nodes still have a zero dimensional origin which then is not properly checked.

**Upgrade instructions**

None

**Review instructions**

None

**Checklist**

- [x] Code follows the PSR-2 coding style
- [x] Tests have been created, run and adjusted as needed
- [x] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

### CODE REVIEW

File: Neos.ContentRepository.BehavioralTests/Tests/Behavior/Features/02-NodeCreation/02-CreateNodeAggregateWithNode_ConstraintChecks_WithDimensions.feature [hunk 1]
```diff
@@ -63,3 +63,22 @@ Feature: Create node aggregate with node
...
+      | parentNodeAggregateId     | "lady-eleonode-rootford"              |
+      | originDimensionSpacePoint | {"language":"de"}                     |
+      | nodeName                  | "document"                            |
+    And the graph projection is fully up to date
+    And the command CreateNodeAggregateWithNode is executed with payload and exceptions are caught:
+      | Key                       | Value                                 |
+      | nodeAggregateId           | "nody-mc-nodeface"                    |
+      | nodeTypeName              | "Neos.ContentRepository.Testing:Node" |
+      | parentNodeAggregateId     | "lady-eleonode-rootford"              |
+      | originDimensionSpacePoint | {"language":"de"}                     |
```
Comment (mhsdesign) [inline Neos.ContentRepository.BehavioralTests/Tests/Behavior/Features/02-NodeCreation/02-CreateNodeAggregateWithNode_ConstraintChecks_WithDimensions.feature:82]: shouldnt be that something different? But i checked it .. it doest matter the test fail before either way and now pass either way.

```suggestion
      | originDimensionSpacePoint | {"language":"mul"}                    |
```

Comment (nezaniel) [reply Neos.ContentRepository.BehavioralTests/Tests/Behavior/Features/02-NodeCreation/02-CreateNodeAggregateWithNode_ConstraintChecks_WithDimensions.feature:82]: mul conflicts also in de, its specialization, thus this makes no difference. 
An additional test case would be create in mul, delete all specializations, then create in de. But that is out of scope here as it requires the deletion feature

Comment (mhsdesign) [reply Neos.ContentRepository.BehavioralTests/Tests/Behavior/Features/02-NodeCreation/02-CreateNodeAggregateWithNode_ConstraintChecks_WithDimensions.feature:82]: ups sorry i was on the wrong path i dont know what i thought ^^

**Review by mhsdesign — APPROVED** [review]
Looks good by reading and i also experimented with the testcase locally a bit and it failed previously.

**Pull request merged**